includes/Product/Shopify_Query.php

```diff
@@ -0,0 +1,239 @@
+<?php
+/**
+ * Class Shopify_Query
+ *
+ * @link      https://github.com/googleforcreators/web-stories-wp
+ *
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ */
+
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Product;
+
+use Google\Web_Stories\Model\Image;
+use Google\Web_Stories\Model\Product;
+use Google\Web_Stories\Model\Rating;
+use Google\Web_Stories\Settings;
+use WP_Error;
+use WP_Http;
+
+/**
+ * Class Shopify_Query
+ *
+ * @phpstan-type ShopifyGraphQLError array{message: string, extensions: array{code: string, requestId: string}}[]
+ * @phpstan-type ShopifyGraphQLPriceRange array{minVariantPrice: array{amount: int, currencyCode: string}}
+ * @phpstan-type ShopifyGraphQLProductImage array{url: string, altText: string}
+ * @phpstan-type ShopifyGraphQLProduct array{id: string, handle: string, title: string, vendor: string, description: string, onlineStoreUrl?: string, images: array{edges: array{node: ShopifyGraphQLProductImage}[]}, priceRange: ShopifyGraphQLPriceRange}
+ * @phpstan-type ShopifyGraphQLResponse array{errors?: ShopifyGraphQLError, data: array{products: array{edges: array{node: ShopifyGraphQLProduct}[]}}}
+ */
+class Shopify_Query extends Query {
+
+	/**
+	 * Settings instance.
+	 *
+	 * @var Settings Settings instance.
+	 */
+	private $settings;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Settings $settings Settings instance.
+	 */
+	public function __construct( Settings $settings ) {
+		$this->settings = $settings;
+	}
+
+	/**
+	 * Get products by search term.
+	 *
+	 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 *
+	 * @since 1.20.0
+	 *
+	 * @param string $search_term Search term.
+	 * @return void|WP_Error
+	 */
+	public function do_search( string $search_term ) {
+
+		/**
+		 * Filters the shopify products data TTL value.
+		 *
+		 * @since 1.20.0
+		 *
+		 * @param int $time Time to live (in seconds). Default is 1 day.
+		 */
+		$cache_ttl = apply_filters( 'web_stories_shopify_data_cache_ttl', DAY_IN_SECONDS ); // TODO: lower default TTL.
+		$cache_key = ! empty( $search_term ) ? 'web_stories_shopify_data_' . md5( $search_term ) : 'web_stories_shopify_data_default';
+
+		$data = get_transient( $cache_key );
+
+		if ( \is_string( $data ) && ! empty( $data ) ) {
```

Comment:
Do we have a way to avoid and or clear the cache when manually testing / troubleshooting --  if not maybe add something here?

Reply:
This is just a copy and paste existing code. We can mock method in classes, so this should not be that much of an issue.